cargo/checksum_test.go

```diff
@@ -0,0 +1,54 @@
+package cargo_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+)
+
+func testChecksum(t *testing.T, context spec.G, it spec.S) {
+	Expect := NewWithT(t).Expect
+
+	context("Matching", func() {
+		type testCaseType struct {
+			c1     string
+			c2     string
+			result bool
+		}
+
+		for _, tc := range []testCaseType{
+			{"", "", true},
+			{"c", "c", true},
+			{"c", "c", true},
+			{"sha256:c", "c", true},
+			{"c", "sha256:c", true},
+			{"md5:c", "md5:c", true},
+			{"md5:c", ":c", false},
+			{":", ":", true},
+			{":c", ":c", true},
+			{"", "c", false},
+			{"c", "", false},
+			{"c", "z", false},
+			{"md5:c", "sha256:c", false},
```

Comment:
For completeness, perhaps it also makes sense to check when the alg is specified the same and the hash is different:
```suggestion
			{"md5:c", "sha256:c", false},
			{"md5:c", "md5:d", false},
```